File: Clocktower.Server/Data/VoteHistoryRecord.cs
namespace Clocktower.Server.Data;

public record VoteHistoryRecord
{
    public DateTime Time { get; set; }
    public string? NominatorId { get; set; }
    public string? NomineeId { get; set; }
    public int VoteCount { get; set; }
    public int RequiredMajority { get; set; }
    public IEnumerable<string> Voters { get; set; } = [];
}

File: Clocktower-UI/src/types/voteHistoryRecord.ts
export class VoteHistoryRecord {
    readonly time: string;
    readonly nominatorId: string;
    readonly nomineeId: string;
    readonly voteCount: number;
    readonly requiredMajority: number;
    readonly voters: string[];

    constructor(record: VoteHistoryRecord) {
        this.time = record.time;
        this.nominatorId = record.nominatorId;
        this.nomineeId = record.nomineeId;
        this.voteCount = record.voteCount;
        this.requiredMajority = record.requiredMajority;
        this.voters = record.voters;
    }

    public get dateTime(): Date {
        return new Date(this.time);
    }
}